admin-base/ui.apps/src/main/content/jcr_root/apps/field/collection/template.vue

```diff
@@ -23,216 +23,257 @@
+    },
+    prepModel(model, schema) {
+      for (let i = 0; i < schema.fields.length; i++) {
+        const field = schema.fields[i].model
+        if (!model[field]) {
+          Vue.set(model, field, '')
         }
-        return false
-      },
-      itemName(item, index) {
-          if(this.schema.fieldLabel) {
-              const len = this.schema.fieldLabel.length
-              for(let i=0; i<len; i++){
-                  let label = this.schema.fieldLabel[i]
-                  let childItem = this.value[index]
-                  if(childItem[label]){
-                      return childItem[label]
-                  }
-              }
-          }
-          return parseInt(index) + 1
-      },
-      prepModel(model, schema) {
-            for(let i = 0; i < schema.fields.length; i++) {
-                const field = schema.fields[i].model
-                if(!model[field]) {
-                    Vue.set(model, field, '')
-                }
-            }
-            return model
-        },
+      }
+      return model
+    },
 
-      onAddItem(e){
-        if(!this.schema.multifield){
-          var newChild = '';
-            if(!this.value || this.value === '')
-            {
-                Vue.set(this, 'value', new Array())
-            }
-        } else {
-            var newChild = { name: 'n' + Date.now()}
-            this.prepModel(newChild, this.schema)
-            newChild['sling:resourceType'] = this.schema.resourceType
-        }
-        if (!this.value) {
-          this.value = []
+    onAddItem(e) {
+      if (!this.schema.multifield) {
+        var newChild = ''
+        if (!this.value || this.value === '') {
+          Vue.set(this, 'value', new Array())
         }
-        this.value.push(newChild)
-        // Vue.set(this.value, this.value.length -1, newChild)
-        this.onSetActiveItem(this.value.length - 1)
-        this.$forceUpdate()
-      },
-      onRemoveItem(item, index){
-        this.value.splice(index, 1)
-        if( this.schema.multifield ) {
-          if( item.hasOwnProperty('path') ) {
-            let _deleted = $perAdminApp.getNodeFromViewWithDefault("/state/tools/_deleted", {});
-            let copy = JSON.parse(JSON.stringify(item));
-            copy._opDelete = true;
-            if(!_deleted[this.schema.model]) _deleted[this.schema.model] = [];
-            _deleted[this.schema.model].push(copy)
-          }
-          this.activeItem = null
+      } else {
+        var newChild = {name: 'n' + Date.now()}
+        this.prepModel(newChild, this.schema)
+        newChild['sling:resourceType'] = this.schema.resourceType
+      }
+      if (!this.value) {
+        this.value = []
+      }
+      this.value.push(newChild)
+      // Vue.set(this.value, this.value.length -1, newChild)
+      this.onSetActiveItem(this.value.length - 1)
+      this.$forceUpdate()
+    },
+    onRemoveItem(item, index) {
+      this.value = this.value.filter((item, i) => i !== index)
```

Comment:
use `filter()` instead of `splice()` to gurantee a dom update due to vuejs reactivity behaviour